x-pack/plugins/maps/public/classes/sources/mvt_single_layer_vector_source/mvt_field_config_editor.tsx

```diff
@@ -0,0 +1,211 @@
+      defaultMessage: 'number',
+    }),
+  },
+].map(makeOption);
+
+export interface Props {
+  fields: MVTFieldDescriptor[];
+  onChange: (fields: MVTFieldDescriptor[]) => void;
+}
+
+interface State {
+  currentFields: MVTFieldDescriptor[];
+}
+
+export class MVTFieldConfigEditor extends Component<Props, State> {
+  state: State = {
+    currentFields: _.cloneDeep(this.props.fields),
+  };
+
+  _notifyChange = _.debounce(() => {
+    const invalid = this.state.currentFields.some((field: MVTFieldDescriptor) => {
+      return field.name === '';
+    });
+
+    if (!invalid) {
+      this.props.onChange(this.state.currentFields);
+    }
+  });
+
+  _fieldChange(newFields: MVTFieldDescriptor[]) {
+    this.setState(
+      {
+        currentFields: newFields,
+      },
+      this._notifyChange
+    );
+  }
+
+  _removeField(index: number) {
+    const newFields: MVTFieldDescriptor[] = this.state.currentFields.slice();
+    newFields.splice(index, 1);
+    this._fieldChange(newFields);
+  }
+
+  _addField = () => {
+    const newFields: MVTFieldDescriptor[] = this.state.currentFields.slice();
+    newFields.push({
+      type: MVT_FIELD_TYPE.STRING,
+      name: '',
+    });
+    this._fieldChange(newFields);
+  };
+
+  _renderFieldTypeDropDown(mvtFieldConfig: MVTFieldDescriptor, index: number) {
+    const onChange = (type: MVT_FIELD_TYPE) => {
+      const newFields = this.state.currentFields.slice();
+      newFields[index] = {
+        type,
+        name: newFields[index].name,
+      };
+      this._fieldChange(newFields);
+    };
+
+    const addButton = (
+      <EuiButtonIcon
+        iconType="trash"
+        color="danger"
+        onClick={() => {
+          this._removeField(index);
+        }}
+        title={i18n.translate('xpack.maps.mvtSource.trashButtonTitle', {
+          defaultMessage: 'Remove field',
+        })}
+        aria-label={i18n.translate('xpack.maps.mvtSource.trashButtonAriaLabel', {
+          defaultMessage: 'Remove field',
+        })}
+      />
+    );
+    return (
+      <EuiSuperSelect
+        options={FIELD_TYPE_OPTIONS}
+        valueOfSelected={mvtFieldConfig.type}
+        onChange={(value) => onChange(value)}
+        append={addButton}
+      />
+    );
+  }
+
+  _renderFieldNameInput(mvtFieldConfig: MVTFieldDescriptor, index: number) {
+    const onChange = (e: ChangeEvent<HTMLInputElement>) => {
+      const name = e.target.value;
+      const newFields = this.state.currentFields.slice();
+      newFields[index] = {
+        name,
+        type: newFields[index].type,
+      };
+      this._fieldChange(newFields);
+    };
+
+    const emptyName = mvtFieldConfig.name === '';
+    let hasDupes = false;
+    for (let i = 0; i < this.state.currentFields.length; i++) {
+      if (i !== index && mvtFieldConfig.name === this.state.currentFields[i].name) {
+        hasDupes = true;
+        break;
+      }
+    }
+    const isInvalid = emptyName || hasDupes;
+    const placeholderText = isInvalid
```

Comment:
Looks like removing this logic was undone. Do not need logic to set placeholder text to empty string. Html elements will take care of this.